--- a/src/SpecObjectInterface.php	2026-04-24 13:10:20
+++ b/src/SpecObjectInterface.php	2026-04-24 13:10:20
@@ -40,7 +40,7 @@
     /**
      * Resolves all Reference Objects in this object and replaces them with their resolution.
      */
-    public function resolveReferences(ReferenceContext $context = null);
+    public function resolveReferences(?ReferenceContext $context = null);
 
     /**
      * Set context for all Reference Objects in this object.
--- a/src/SpecBaseObject.php	2026-04-24 13:10:20
+++ b/src/SpecBaseObject.php	2026-04-24 13:10:20
@@ -393,7 +393,7 @@
      * Resolves all Reference Objects in this object and replaces them with their resolution.
      * @throws exceptions\UnresolvableReferenceException in case resolving a reference fails.
      */
-    public function resolveReferences(ReferenceContext $context = null)
+    public function resolveReferences(?ReferenceContext $context = null)
     {
         // avoid recursion to get stuck in a loop
         if ($this->_recursingReferences) {
--- a/src/spec/Paths.php	2026-04-24 13:10:20
+++ b/src/spec/Paths.php	2026-04-24 13:10:20
@@ -246,7 +246,7 @@
      * Resolves all Reference Objects in this object and replaces them with their resolution.
      * @throws UnresolvableReferenceException
      */
-    public function resolveReferences(ReferenceContext $context = null)
+    public function resolveReferences(?ReferenceContext $context = null)
     {
         foreach ($this->_paths as $key => $path) {
             if ($path === null) {
--- a/src/spec/PathItem.php	2026-04-24 13:10:20
+++ b/src/spec/PathItem.php	2026-04-24 13:10:20
@@ -150,7 +150,7 @@
      * Resolves all Reference Objects in this object and replaces them with their resolution.
      * @throws \cebe\openapi\exceptions\UnresolvableReferenceException in case resolving a reference fails.
      */
-    public function resolveReferences(ReferenceContext $context = null)
+    public function resolveReferences(?ReferenceContext $context = null)
     {
         if ($this->_ref instanceof Reference) {
             $pathItem = $this->_ref->resolve($context);
--- a/src/spec/Responses.php	2026-04-24 13:10:20
+++ b/src/spec/Responses.php	2026-04-24 13:10:20
@@ -236,7 +236,7 @@
      * Resolves all Reference Objects in this object and replaces them with their resolution.
      * @throws UnresolvableReferenceException
      */
-    public function resolveReferences(ReferenceContext $context = null)
+    public function resolveReferences(?ReferenceContext $context = null)
     {
         foreach ($this->_responses as $key => $response) {
             if ($response instanceof Reference) {
--- a/src/spec/Callback.php	2026-04-24 13:10:20
+++ b/src/spec/Callback.php	2026-04-24 13:10:20
@@ -133,7 +133,7 @@
      * Resolves all Reference Objects in this object and replaces them with their resolution.
      * @throws UnresolvableReferenceException
      */
-    public function resolveReferences(ReferenceContext $context = null)
+    public function resolveReferences(?ReferenceContext $context = null)
     {
         if ($this->_pathItem !== null) {
             $this->_pathItem->resolveReferences($context);
--- a/src/spec/Reference.php	2026-04-24 13:10:20
+++ b/src/spec/Reference.php	2026-04-24 13:10:20
@@ -69,7 +69,7 @@
      * @param string|null $to class name of the type referenced by this Reference
      * @throws TypeErrorException in case invalid data is supplied.
      */
-    public function __construct(array $data, string $to = null)
+    public function __construct(array $data, ?string $to = null)
     {
         $this->_rawSpec = $data;
         if (!isset($data['$ref'])) {
@@ -176,7 +176,7 @@
      * If you call resolveReferences() make sure to replace the Reference with the resolved object first.
      * @throws UnresolvableReferenceException in case of errors.
      */
-    public function resolve(ReferenceContext $context = null)
+    public function resolve(?ReferenceContext $context = null)
     {
         if ($context === null) {
             $context = $this->getContext();
@@ -363,7 +363,7 @@
      * Resolves all Reference Objects in this object and replaces them with their resolution.
      * @throws UnresolvableReferenceException
      */
-    public function resolveReferences(ReferenceContext $context = null)
+    public function resolveReferences(?ReferenceContext $context = null)
     {
         throw new UnresolvableReferenceException('Cyclic reference detected, resolveReferences() called on a Reference Object.');
     }
